src/Akeneo/Pim/Automation/DataQualityInsights/tests/back/Specification/Infrastructure/Connector/Tasklet/MarkEvaluation/MarkProductModelsEvaluationImpactedByUpdatedAttributeGroupActivationTaskletSpec.php

```diff
@@ -0,0 +1,86 @@
+<?php
+
+namespace Specification\Akeneo\Pim\Automation\DataQualityInsights\Infrastructure\Connector\Tasklet\MarkEvaluation;
+
+use Akeneo\Pim\Automation\DataQualityInsights\Application\ProductEvaluation\CreateCriteriaEvaluations;
+use Akeneo\Pim\Automation\DataQualityInsights\Domain\Query\ProductEvaluation\GetEntityIdsImpactedByAttributeGroupActivationQueryInterface;
+use Akeneo\Pim\Automation\DataQualityInsights\Domain\ValueObject\ProductModelIdCollection;
+use Akeneo\Pim\Automation\DataQualityInsights\Infrastructure\Connector\JobParameters\PrepareEvaluationsParameters;
+use Akeneo\Pim\Automation\DataQualityInsights\Infrastructure\Connector\Tasklet\MarkEvaluation\MarkProductModelsEvaluationImpactedByUpdatedAttributeGroupActivationTasklet;
+use Akeneo\Tool\Component\Batch\Job\JobParameters;
+use Akeneo\Tool\Component\Batch\Model\JobExecution;
+use Akeneo\Tool\Component\Batch\Model\StepExecution;
+use Akeneo\Tool\Component\Connector\Step\TaskletInterface;
+use PhpSpec\ObjectBehavior;
+use Prophecy\Argument;
+use Psr\Log\LoggerInterface;
+use Webmozart\Assert\Assert;
+
+class MarkProductModelsEvaluationImpactedByUpdatedAttributeGroupActivationTaskletSpec extends ObjectBehavior
+{
+    public function let(
+        CreateCriteriaEvaluations $createCriteriaEvaluations,
+        GetEntityIdsImpactedByAttributeGroupActivationQueryInterface $getProductModelIdsImpactedByAttributeGroupActivationQuery,
+        LoggerInterface $logger,
+    ): void {
+        $this->beConstructedWith($createCriteriaEvaluations, $getProductModelIdsImpactedByAttributeGroupActivationQuery, $logger, 2);
+    }
+
+    public function it_is_a_tasklet(): void
+    {
+        $this->shouldImplement(TaskletInterface::class);
+        $this->shouldHaveType(MarkProductModelsEvaluationImpactedByUpdatedAttributeGroupActivationTasklet::class);
+    }
+
+    public function it_marks_product_models_evaluation_impacted_by_updated_attribute_group_activation(
+        CreateCriteriaEvaluations $createCriteriaEvaluations,
+        GetEntityIdsImpactedByAttributeGroupActivationQueryInterface $getProductModelIdsImpactedByAttributeGroupActivationQuery,
+        LoggerInterface $logger
+    ): void {
+        $stepExecution = $this->buildStepExecution();
+        $this->setStepExecution($stepExecution);
+
+        $productModelIds = [
+            ProductModelIdCollection::fromStrings(['42', '657']),
+            ProductModelIdCollection::fromStrings(['777'])
+        ];
+
+        $getProductModelIdsImpactedByAttributeGroupActivationQuery
+            ->updatedSince(Argument::that(fn (\DateTimeImmutable $updatedSince) => $updatedSince->format('Y-m-d H:i:s') === '2023-02-07 14:23:56'), 2)
+            ->willReturn(new \ArrayIterator($productModelIds));
+
+        $createCriteriaEvaluations->createAll($productModelIds[0])->shouldBeCalled();
+        $createCriteriaEvaluations->createAll($productModelIds[1])->shouldBeCalled();
```

Comment:
```suggestion
        $createCriteriaEvaluations->createAll($productModelIds[0])->shouldBeCalledOnce();
        $createCriteriaEvaluations->createAll($productModelIds[1])->shouldBeCalledOnce();
```

Because if it's called more then once it's not good.